src/compiler/transformers/es2015.ts

```diff
@@ -3658,16 +3678,21 @@ namespace ts {
                 }
             }
 
-            // NOTE: The parentheses here is entirely optional as we are now able to auto-
-            //       parenthesize when rebuilding the tree. This should be removed in a
-            //       future version. It is here for now to match our existing emit.
-            return createParen(
-                inlineExpressions([
-                    createAssignment(temp, createArrayLiteral(cookedStrings)),
-                    createAssignment(createPropertyAccess(temp, "raw"), createArrayLiteral(rawStrings)),
-                    createCall(tag, /*typeArguments*/ undefined, templateArguments)
-                ])
-            );
+            const helperCall = createTemplateObjectHelper(context, createArrayLiteral(cookedStrings), createArrayLiteral(rawStrings));
+
+            // If we're in the global scope, we risk having conflicting variables.
+            // Since we currently lack the infrastructure to create sufficiently unique names,
+            // we'll fall back to creating the template object on every invocation.
+            templateArguments[0] = !temp ?

```

Comment:
if isExternalModule is cheap to call, it would be nice to do so here; much easier to read.

Reply:
Also consider flipping the conditional.